rechnung/contract.py

```diff
@@ -19,8 +19,9 @@ def get_contracts(settings, year=None, month=None, cid_only=None, inactive=False
         if cid_only and cid_only != filename.stem:
             continue
 
-        with open(Path(settings.contracts_dir / filename), "r") as contract_file:
-            contract = yaml.safe_load(contract_file)
+        contract = yaml.safe_load(
+            (settings.contracts_dir / filename).read_text("utf-8")
+        )
```

Comment:
Reading everything into a buffer to then passing it to `yaml.safe_load` seems rather inelegant, as `yaml.safe_load` also accepts a stream.

Instead of using the `read_text` method of pathlib and duplicating everything in memory, the encoding problems could probably also be solved like this:
```suggestion
        with open(Path(settings.contracts_dir / filename), "r", "utf-8") as contract_file:
            contract = yaml.safe_load(contract_file)
```

Reply:
After all I think the encoding was partly a problem on my side. I think some Python files where for whatever reason in a different encoding causing a lot of trouble. Should I fix this particular issue or should I revert it?